src/state.py

```diff
@@ -87,21 +95,49 @@ def validate_hostname(cls, value: str) -> str:
     def validate_path(cls, value: str) -> str:
         """Validate the path.
 
+        Args:
+            value: The value to validate.
+
+        Returns:
+            The value after validation.
+        """
+        return validate_path_value(value)
+
+    @pydantic.field_validator("backends_path")
+    @classmethod
+    def validate_backends_path(cls, value: str) -> str:
+        """Validate the backends_path.
+
+        Args:
+            value: The value to validate.
+
+        Returns:
+            The value after validation.
+        """
+        return validate_path_value(value)
+
+    @pydantic.field_validator("proxy_cache_valid")
+    @classmethod
+    def validate_proxy_cache_valid(cls, value: tuple[str, ...]) -> tuple[str, ...]:
+        """Validate the proxy_cache_valid.
+
         Args:
             value: The value to validate.
 
         Raises:
-            ValueError: Error in validation.
+            ValueError: The proxy_cache_valid is invalid.
 
         Returns:
             The value after validation.
         """
-        # This are the valid characters for path in addition to `/`:
-        # a-z A-Z 0-9 . - _ ~ ! $ & ' ( ) * + , ; = : @
-        # https://datatracker.ietf.org/doc/html/rfc3986#section-3.3
-        valid_path = re.compile(r"[/\w.\-~!$&'()*+,;=:@]+", re.IGNORECASE)
-        if valid_path.fullmatch(value) is None:
-            raise ValueError("Path contains non-allowed character")
+        for item in value:
+            tokens = item.split(" ")
+            if len(tokens) < 2:
+                raise ValueError(f"Invalid item in proxy_cache_valid: {item}")
```

Comment:
We should probably be more specific on what is missing leading to the exception, to help the user if they fall into that error

Reply:
Global comment for every ValueError in the pydantic validators